app/models/planning_application.rb

```diff
@@ -112,7 +113,15 @@ def days_overdue
   end
 
   def reference
-    @reference ||= id.to_s.rjust(8, "0")
+    @reference ||= "#{created_at_year}-#{application_number}-#{application_type_code}"
+  end
+
+  def reference_in_full
+    @reference_in_full ||= "#{local_authority.council_code}-#{reference}"
+  end
+
+  def application_number
+    super.to_s.rjust(5, "0")
```

Comment:
I wonder if `self[:application_number]` would be clearer than `super`? I think that way it's more obvious that this is overwriting a getter method. https://github.com/rubocop/rails-style-guide#read-attribute

Reply:
Ah okay, thanks for sharing. I have no strong preference for this so if it helps makes thing clearer then will update

Reply:
Ah okay, thanks for sharing. I have no strong preference for this so if it helps makes thing clearer then will update